packages/wallets/phantom/src/adapter.ts

```diff
@@ -109,40 +136,54 @@ export class PhantomWalletAdapter extends BaseMessageSignerWalletAdapter {
         return this._readyState;
     }
 
-    async connect(): Promise<void> {
-        try {
-            if (this.connected || this.connecting) return;
-            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
-
-            this._connecting = true;
-
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            const wallet = window.phantom?.solana || window.solana!;
-
-            if (!wallet.isConnected) {
-                try {
-                    await wallet.connect();
-                } catch (error: any) {
-                    throw new WalletConnectionError(error?.message, error);
-                }
-            }
+    async connectWithWallet(): Promise<void> {
```

Comment:
We shouldn't add new public methods like this. It should be `#private`, but the name is a bit confusing.